lib/ansible/modules/crypto/luks_device.py

```diff
@@ -193,10 +231,41 @@ class Handler(object):
     def __init__(self, module):
         self._module = module
         self._lsblk_bin = self._module.get_bin_path('lsblk', True)
+        self._blkid_bin = self._module.get_bin_path('blkid', True)
 
     def _run_command(self, command):
         return self._module.run_command(command)
 
+    def get_device_by_uuid(self, uuid):
+        ''' Returns the device that holds UUID passed by user
+        '''
+        uuid = self._module.params['uuid']
+        if uuid is None:
+            return None
+
+        query = '%s=%s' % ('UUID', uuid)
+        result = self._run_command([self._blkid_bin, '-l', '-t', query])
+
+        if result[RETURN_CODE] != 0:
+            return None
+
+        return result[STDOUT].split(':')[0]
+
+    def get_device_by_label(self, label):
+        ''' Returns the device that holds label passed by user
+        '''
+        label = self._module.params['label']
+        if label is None:
+            return None
+
+        query = '%s=%s' % ('LABEL', label)
+        result = self._run_command([self._blkid_bin, '-l', '-t', query])
```

Comment:
Probably the same with `--label` instead of `--uuid` here.

Reply:
Nice, much better sir, thanks ;-D